### PR TITLE
Expose the registered device ID from the `register_appservice_user` test helper.

### PR DESCRIPTION
**Pull request by reivilibre**
This is part of the E2E AS work, spun out onto develop for sanity's sake.

### CODE REVIEW

File: tests/unittest.py [hunk 1]
```diff
@@ -620,18 +620,18 @@ def register_appservice_user(
...
         Args:
             username: the user to be registered by an application service.
-                Should be a full username, i.e. ""@localpart:hostname" as opposed to just "localpart"
+                Should NOT be a full username, i.e. just "localpart" as opposed to "@localpart:hostname"
             appservice_token: the acccess token for that application service.
 
         Raises: if the request to '/register' does not return 200 OK.
 
-        Returns: the MXID of the new user.
+        Returns: the MXID of the new user, the device ID of the new user's first device.
```
Comment (clokep) [inline tests/unittest.py:634]:
```suggestion
        Returns:
            The MXID of the new user, the device ID of the new user's first device.
```

**Review by clokep — APPROVED** [review]
Seems reasonable. I'll take your word that this is used in upcoming tests!

File: tests/unittest.py (hunk 1 above)
Comment (reivilibre) [reply tests/unittest.py:634]: ah deary me how did I forget to update that. Thanks! :)

**Comment by anoadragon453** [comment]
@reivilibre is merging this PR blocked on anything?

**Comment by reivilibre** [comment]
> @reivilibre is merging this PR blocked on anything?

No, other than the PR that needs it is a while away (blocked on yours) and it felt a bit presumptuous to put the test there so much earlier than the PR itself.

**Pull request merged**